ColorPicker/IkColorPicker.swift

```diff
@@ -0,0 +1,243 @@
+        
+        return hueImage;
+    }
+
+    func createFullColorImage(color :UIColor) -> UIImage
+    {
+        let imageHeight: CGFloat = 40;
+        let imageWidth: CGFloat  = 40;
+        let size: CGSize = CGSize(width: imageWidth, height: CGFloat(imageHeight));
+
+        UIGraphicsBeginImageContextWithOptions(size, false, 0);
+        let context: CGContextRef = UIGraphicsGetCurrentContext();
+        let rec = CGRect(x: 0, y: 0, width: size.width , height: size.height);
+
+        CGContextSetFillColorWithColor(context, color.CGColor);
+        let roundedRect = UIBezierPath(roundedRect: rec, cornerRadius: 6);
+        roundedRect.fillWithBlendMode(kCGBlendModeNormal, alpha: 1);
+        
+        let fullColorImage =  UIGraphicsGetImageFromCurrentImageContext();
+    
+        UIGraphicsEndImageContext();
+    
+        return fullColorImage;
+    }
+    
+    func loadView()
+    {
+        self.frame = CGRect(x:0,y:0,width:320,height:viewPickerHeight);
+        
+        let hueColorImage = self.createHueColorImage();
+        self.hueColorsImageView = UIImageView(image: hueColorImage);
+        self.addSubview(self.hueColorsImageView!);
+        
+        let panRecognizer = UIPanGestureRecognizer(target: self, action:"baseColorPicking:");
+        self.hueColorsImageView!.addGestureRecognizer(panRecognizer);
+        
+        let tapRecognizer = UITapGestureRecognizer(target: self, action: "baseColorPicking:");
+        self.hueColorsImageView!.addGestureRecognizer(tapRecognizer);
+        self.hueColorsImageView!.userInteractionEnabled = true;
+        
+        
+        //------
+        let brightnessColorImage = self.createBrightnessImage(self.selectedBaseColor);
+        self.brightnessColorsImageView = UIImageView(image: brightnessColorImage);
+        self.addSubview(self.brightnessColorsImageView!);
+    
+        var brImgRect = self.brightnessColorsImageView!.frame;
+        brImgRect.origin.y = self.hueColorsImageView!.frame.origin.y + self.hueColorsImageView!.frame.size.height;
+        self.brightnessColorsImageView!.frame = brImgRect;
+        self.brightnessColorsImageView!.userInteractionEnabled = true;
+    
+        let brightSlideGesture = UIPanGestureRecognizer(target: self, action:Selector("colorPicking:"));
+        self.brightnessColorsImageView!.addGestureRecognizer(brightSlideGesture);
+        
+        let brightTapGesture = UITapGestureRecognizer(target: self, action: Selector("colorPicking:"));
+        self.brightnessColorsImageView!.addGestureRecognizer(brightTapGesture);
+        self.brightnessColorsImageView!.userInteractionEnabled = true;
+    }
+    
+    internal var selectedColor: UIColor
+    {
+        didSet{
+            if((self.fullColorImageView) != nil)
+            {
+                let fullColorImage = self.createFullColorImage(selectedColor);
+                fullColorImageView!.image = fullColorImage;
+            }
+            
+            if(self.delegate != nil)
```

Comment:
you could instead say self.delegate?.colorSelectedChanged(self.selectedColor) instead of checking for nil